src/components/TodoList.js

```diff
@@ -18,14 +18,38 @@ const TodoList = ({ todos, deleteTodo, completeTodo }) => {
     deleteTodo(id);
   };
 
-  const listItems = todos.map(todo => (
+  let filteredTodos;
+  switch (activeTab) {
+    case 0:
+      filteredTodos = todos.filter(
+        todo => todo.isCompleted === false && todo.isExpired === false
+      );
+      break;
+    case 1:
+      filteredTodos = todos.filter(todo => todo.isCompleted === true);
+      break;
+    case 2:
+      filteredTodos = todos.filter(
+        todo => todo.isCompleted === false && todo.isExpired === true
```

Comment:
this is interesting, lets refactor it.

Reply:
I replaced the index numbers with explicit strings for active, completed, and missed todos.